photon-client/src/main/scala/com/linkedin/photon/ml/data/avro/ModelProcessingUtils.scala

```diff
@@ -159,53 +153,96 @@ object ModelProcessingUtils {
     // Load the fixed effect model(s)
     val fixedEffectModelInputDir = new Path(modelsDir, AvroConstants.FIXED_EFFECT)
     val fixedEffectModels = if (fs.exists(fixedEffectModelInputDir)) {
-      fs.listStatus(fixedEffectModelInputDir).map { fileStatus =>
-        val innerPath = fileStatus.getPath
-        val name = innerPath.getName
-
-        // Load the model ID info
-        val idInfoPath = new Path(innerPath, AvroConstants.ID_INFO)
-        val Array(featureShardId) = IOUtils.readStringsFromHDFS(idInfoPath, configuration).toArray
-        require(featureShardId != null && !featureShardId.isEmpty)
-
-        // Load the coefficients
-        val indexMapLoaders = featureShardIdToIndexMapLoader.flatMap(_.get(featureShardId))
-        val featureNameAndTermToIndexMap = indexMapLoaders.map(_.indexMapForDriver())
-        val modelPath = new Path(innerPath, AvroConstants.COEFFICIENTS)
-        val (glm, featureIndexLoader) = loadGLMFromHDFS(modelPath.toString, featureNameAndTermToIndexMap, sc)
-
-        (name, featureIndexLoader, new FixedEffectModel(sc.broadcast(glm), featureShardId))
-      }
+
+      fs
+        .listStatus(fixedEffectModelInputDir)
+        .map(_.getPath)
+        .filter { innerPath =>
+          coordinatesToLoadOpt match {
+            case Some(coordinatesToLoad) => coordinatesToLoad.contains(innerPath.getName)
+            case None => true
+          }
+        }
+        .map { innerPath =>
+
+          val name = innerPath.getName
+
+          // Load the model ID info
+          val idInfoPath = new Path(innerPath, AvroConstants.ID_INFO)
+          val Array(featureShardId) = IOUtils.readStringsFromHDFS(idInfoPath, configuration).toArray
+
+          require(
+            featureShardId != null && !featureShardId.isEmpty,
```

Comment:
Why would `featureShardId` be null? If we don't get a match on line 172, it will result in `MatchError`. Is it possible that `IOUtils.readStringsFromHDFS` would return something like `Array(null)`? On a quick reading of the function, it doesn't appear so.

Reply:
I think I'm going to rewrite the check instead, to explicitly check for metadata input errors.

Reply:
Or maybe not - either way, you're right.